runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/config/HttpBindingConfig.java

```diff
@@ -252,173 +261,142 @@ public HttpRequestType resolveRequest(
-                    String value = nextAmpersandIndex == -1 ?
-                        query.substring(equalsIndex + 1) :
-                        query.substring(equalsIndex + 1, nextAmpersandIndex);
-                    if (!key.isEmpty())
-                    {
-                        queryParams.put(URLDecoder.decode(key, UTF_8), new String8FW(URLDecoder.decode(value, UTF_8)));
-                    }
-                }
-                ampersandIndex = nextAmpersandIndex;
-            }
-            while (ampersandIndex != -1);
-        }
-        return queryParams;
+        return requestType.pathMatcher.reset(path).matches();
     }
 
     public boolean validateHeaders(
-        HttpRequestType request,
+        HttpRequestType requestType,
         HttpBeginExFW beginEx)
     {
-        return request == null ||
-            validateHeaderValues(request, beginEx) &&
-            validatePathParams(request) &&
-            validateQueryParams(request);
+        String path = beginEx.headers().matchFirst(h -> h.name().equals(HEADER_PATH)).value().asString();
+        return requestType == null ||
+            validateHeaderValues(requestType, beginEx) &&
+            validatePathParams(requestType, path) &&
+            validateQueryParams(requestType, path);
     }
 
     private boolean validateHeaderValues(
-        HttpRequestType request,
+        HttpRequestType requestType,
         HttpBeginExFW beginEx)
     {
-        MutableBoolean isValid = new MutableBoolean(true);
-        if (request != null && request.headers != null)
+        MutableBoolean valid = new MutableBoolean(true);
+        if (requestType != null && requestType.headers != null)
         {
             beginEx.headers().forEach(header ->
             {
-                if (isValid.get())
+                if (valid.value)
                 {
-                    Validator validator = request.headers.get(header.name());
+                    Validator validator = requestType.headers.get(header.name());
                     if (validator != null)
                     {
                         String16FW value = header.value();
                         if (!validator.read(value.value(), value.offset(), value.length()))
                         {
-                            isValid.set(false);
+                            valid.value = false;
                         }
                     }
                 }
             });
         }
-        return isValid.get();
+        return valid.value;
     }
 
     private boolean validatePathParams(
-        HttpRequestType request)
+        HttpRequestType requestType,
+        String path)
     {
-        boolean isValid = true;
-        for (String name : request.pathParamValues.keySet())
+        Matcher matcher = requestType.pathMatcher.reset(path);
+        boolean matches = matcher.matches();
+        assert matches;
+
+        boolean valid = true;
+        for (String name : requestType.pathParams.keySet())
```

Comment:
Make sure we are not creating a new iterator every time.
See `Object2ObjectHashMap` if needed.

Reply:
I replaced `HashMap` with `Object2ObjectHashMap`, LGTM.